tests/chainer_tests/functions_tests/loss_tests/test_absolute_error.py

```diff
@@ -26,74 +26,89 @@
+        return (x0, x1)
 
-        for i in numpy.ndindex(self.x0.shape):
-            # Compute expected value
-            loss_expect = abs(self.x0[i] - self.x1[i])
-            assert round(loss_value[i] - loss_expect, 5) == 0
+    def forward_expected(self, inputs):
+        x0, x1 = inputs
+        return utils.force_array(numpy.abs(x0 - x1), self.dtype),
 
-    def test_forward_cpu(self):
-        self.check_forward(self.x0, self.x1)
+    def forward(self, inputs, device):
+        x0, x1 = inputs
+        return functions.absolute_error(x0, x1),
 
-    @attr.gpu
-    def test_forward_gpu(self):
-        self.check_forward(cuda.to_gpu(self.x0), cuda.to_gpu(self.x1))
 
-    def check_backward(self, x0_data, x1_data, y_grad):
-        gradient_check.check_backward(
-            functions.absolute_error,
-            (x0_data, x1_data), y_grad, dtype='d')
-
-    def test_backward_cpu(self):
-        self.check_backward(self.x0, self.x1, self.gy)
-
-    @attr.gpu
-    def test_backward_gpu(self):
-        self.check_backward(
-            cuda.to_gpu(self.x0), cuda.to_gpu(self.x1), cuda.to_gpu(self.gy))
-
-    def check_double_backward(self, x0_data, x1_data, y_grad,
-                              gx0_grad, gx1_grad):
-        gradient_check.check_double_backward(
-            functions.absolute_error, (x0_data, x1_data), y_grad,
-            (gx0_grad, gx1_grad), eps=1e-2, **self.double_backward_options)
+@testing.parameterize(*testing.product_dict(
+    [
+        {'dtype': numpy.float16},
+        {'dtype': numpy.float32},
+        {'dtype': numpy.float64},
+    ],
+    [
+        {'shape': (4, 3)},
+        {'shape': (4, 3, 2)},
+        {'shape': (4,)},
+        {'shape': ()},
+        {'shape': (1,)},
+        {'shape': (1, 1)},
+    ]
+))
+@testing.inject_backend_tests(
+    'test_backward_non_default_gpu',
+    testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never', 'always'],
+    })
+)
+class TestNonDefaultGPU(unittest.TestCase):
 
-    def test_double_backward_cpu(self):
-        self.check_double_backward(
-            self.x0, self.x1, self.gy, self.ggx0, self.ggx1)
+    # This test is for https://github.com/chainer/chainer/issues/4669
 
-    @attr.gpu
-    def test_double_backward_gpu(self):
-        self.check_double_backward(
-            cuda.to_gpu(self.x0), cuda.to_gpu(self.x1), cuda.to_gpu(self.gy),
-            cuda.to_gpu(self.ggx0), cuda.to_gpu(self.ggx1))
+    def setUp(self):
+        self.x0 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        # Add sufficient margin to prevent computational error
+        diff = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        diff[abs(diff) < 0.02] = 0.5
+        self.x1 = numpy.asarray(self.x0 + diff)
+        self.gy = numpy.random.random(self.shape).astype(self.dtype)
 
-    # test for #4669
     @attr.multi_gpu(2)
-    def test_backward_non_default_gpu(self):
+    def test_backward_non_default_gpu(self, backend_config):
```

Comment:
Previously this test did not use `@testing.inject_backend_tests`.
I think there's no strong motivation to run this test for various backends.